test/unit/test_disco_elasticsearch.py

```diff
@@ -0,0 +1,311 @@
+"""
+Tests of disco_elasticache
+"""
+import random
+import json
+
+from unittest import TestCase
+from mock import MagicMock
+from disco_aws_automation import DiscoElasticsearch
+from disco_aws_automation.disco_aws_util import is_truthy
+from test.helpers.patch_disco_aws import get_mock_config
+
+MOCK_AWS_CONFIG_DEFINITON = {
+    "disco_aws": {
+        "default_domain_name": "aws.example.com",
+        "http_proxy_hostclass": "mhcproxy"
+    },
+    "mhcproxy": {
+        "eip": "192.0.2.0"
+    }
+}
+
+MOCK_ES_CONFIG_DEFINITON = {
+    "foo:logs": {
+        "instance_type": "m3.medium.elasticsearch",
+        "instance_count": "3",
+        "dedicated_master": "yes",
+        "zone_awareness": "true",
+        "dedicated_master_type": "m3.medium.elasticsearch",
+        "dedicated_master_count": "1",
+        "ebs_enabled": "true",
+        "volume_type": "io1",
+        "volume_size": "10",
+        "iops": "10000",
+        "snapshot_start_hour": "5",
+        "allowed_source_ips": "192.0.2.100 192.0.2.200"
+    },
+    "foo:other-logs": {
+        "instance_type": "t2.medium.elasticsearch",
+        "instance_count": "1",
+        "ebs_enabled": "True",
+        "volume_type": "standard",
+        "volume_size": "10",
+        "snapshot_start_hour": "8"
+    },
+    "bar:strange-logs": {
+        "instance_type": "t2.medium.elasticsearch",
+        "instance_count": "1",
+        "ebs_enabled": "True",
+        "volume_type": "standard",
+        "volume_size": "10",
+        "snapshot_start_hour": "5"
+    }
+}
+
+
+def _get_mock_route53():
+    route53 = MagicMock()
+    return route53
+
+
+class DiscoElastiSearchTests(TestCase):
```

Comment:
They are being mocked. See line 80 and lines 121-125 of this file.

There is some boto3 logging that gets fired at the start, because it tries to make a connection to boto3 at the start. That could be fixed by passing in the mocked connection object rather than creating it later.